test-lib.sh

```diff
@@ -0,0 +1,142 @@
+#
+# Test a container image.
+#
+# Always use sourced from a specific container testfile 
+#
+# reguires definition of CID_FILE_DIR
+# CID_FILE_DIR=$(mktemp --suffix=<container>_test_cidfiles -d)
+# reguires definition of TEST_LIST 
+# TEST_LIST="\
+# run_container_creation_tests
+# run_doc_test <words_to_look_for_in_the_doc>"
+
+# Container CI tests
+# abbreviated as "concit"
+
+# may be redefined in the specific container testfile
+EXPECTED_EXIT_CODE=0
+
+function concit_cleanup() {
+  for cid_file in $CID_FILE_DIR/* ; do
+    CONTAINER=$(cat $cid_file)
```

Comment:
The CONTAINER variable is not `local`, which creates shell variable pollution.

I would encourage you to document shell functions like:
```bash
# foo_action [ARGS...]
# --------------------
# Does "something" with all ARGS. Set's $X to $Z value.
# Defines: $X
# Uses: $Y
foo_actions ()
{
    X=$Y
    for arg ; do
        something "$arg"
    done
    local Z=baz # Z is not visible outside...
}
```